adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -152,8 +167,27 @@ private void validateAcquireTokenRequest(final AuthenticationRequest authenticat
         }
 
         // validate authority
+        Telemetry.getInstance().startEvent(authenticationRequest.getTelemetryRequestId(),
+                EventStrings.AUTHORITY_VALIDATION_EVENT);
+        APIEvent apiEvent = new APIEvent(EventStrings.AUTHORITY_VALIDATION_EVENT);
```

Comment:
>APIEvent apiEvent [](start = 8, length = 17)

small suggestion, i think you can have a builder for those events, and do a build when all the values are set, and also throw errors to make sure build is not called twice.